### DISCUSSION TITLE
HTTP Stream support is now Generally Available! 🎉

### DISCUSSION THREAD

**Discussion by ejizba** [Azure/azure-functions-nodejs-library#261] [Announcements]
Read our [blog post](https://aka.ms/azfuncnodehttpstreams) for more information and use this thread to discuss!

**Comment by ejizba** [comment]
_Originally posted by @eric-gonzalez-tfs in https://github.com/Azure/azure-functions-nodejs-library/issues/97#issuecomment-2123053363_

Hi @ejizba, Thank you for the update!

After calling the `setup` method with `enableHttpStream` set to true, will `await request.formData()` take advantage of the streaming request functionality?

The [blog post](https://techcommunity.microsoft.com/t5/apps-on-azure-blog/azure-functions-support-for-http-streams-in-node-js-is-generally/ba-p/4140209) suggests using `request.body` but is `request.formData` just as good? Thanks!

![image](https://github.com/Azure/azure-functions-nodejs-library/assets/82120345/91dfa1b7-e47b-4017-8a2e-6898aa830ec0)

**Reply by ejizba** [reply]
@eric-gonzalez-tfs I believe it should, but I'll have to whip up some sample code and get back to you on this

**Reply by ejizba** [reply]
@eric-gonzalez-tfs yes you can stream with `formData`! 🎉 [Here](https://github.com/Azure-Samples/azure-functions-nodejs-stream/blob/form/src/functions/streamRequest.ts) is my sample function. The full code is on the "form" branch of our sample http stream repo [here](https://github.com/Azure-Samples/azure-functions-nodejs-stream/tree/form). I modified the `streamRequest` scenario to use a form, so you can follow the instructions in the README for that scenario if you want to try it out

Ironically I had way more difficulty _sending_ the form as a stream. You could theoretically use any tool (curl, browser, postman, etc.) but I found a lot of places buffered it in memory instead of streaming. I eventually found a workaround to use in my node.js script to send it (see [here](https://github.com/Azure-Samples/azure-functions-nodejs-stream/blob/form/src/scripts/streamRequest.ts)).

**Reply by eric-gonzalez-tfs** [reply]
Thanks for following up with these examples. This is super helpful, including the request itself!

Appreciate your quick responses & consistent support @ejizba 🙏

**Comment by ejizba** [comment]
_Originally posted by @georgal in https://github.com/Azure/azure-functions-nodejs-library/issues/97#issuecomment-2123270319_

hey everyone,

as @ilyachenko indicates, the streaming works like a charm, but in my scenario only on localhost. 
Once running from app service it doesn't stream, instead returns response once the stream is completed. 

Is this something anyone witnessed? Is there any way to address the issue? azure configs, specific tier etc? 

Thanks in advance!

**Reply by ejizba** [reply]
Hi @georgal can you give more details about your scenario? For example can you share sample code or your app name using [these instructions](https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately) to do it privately. You hopefully shouldn't need a specific tier or Azure config

**Reply by georgal** [reply]
Follow up here, 

Problem is related to client firewall blockers. 
New release works good! 

Thanks 👍

**Reply by ejizba** [reply]
Awesome, glad you got it figured out!
